docs/literate/src/files/p4est_from_gmsh.jl

```diff
@@ -0,0 +1,452 @@
+#src # `P4estMesh` from [`gmsh`](https://gmsh.info/)
+
+# Trixi.jl supports numerical approximations from structured and unstructured quadrilateral meshes
+# with the [`P4estMesh`](@ref) mesh type.
+
+# The purpose of this tutorial is to demonstrate how to use the `P4estMesh`
+# functionality of Trixi.jl for existing meshes with straight-sided (bilinear) elements/cells.
+# This begins by running and visualizing an available unstructured quadrilateral mesh example.
+# Then, the tutorial will cover how to use existing meshes generated by [`gmsh`](https://gmsh.info/)
+# or any other meshing software that can export to the Abaqus input `.inp` format.
+
+# ## Running the simulation of a near-field flow around an airfoil
+
+# Trixi.jl supports solving hyperbolic problems on several mesh types.
+# A somewhat complex example that employs the `P4estMesh` is the near-field simulation of a
+# Mach 2 flow around the NACA6412 airfoil. 
+
+using Trixi
+redirect_stdio(stdout=devnull, stderr=devnull) do # code that prints annoying stuff we don't want to see here #hide #md
+trixi_include(joinpath(examples_dir(), "p4est_2d_dgsem", "elixir_euler_NACA6412airfoil_mach2.jl"), tspan=(0.0, 0.5))
+end #hide #md
+
+# Conveniently, we can use the Plots package to have a first look at the results:
+# ```julia
+# using Plots
+# pd = PlotData2D(sol)
+# plot(pd["rho"])
+# plot!(getmesh(pd))
+# ```
+
+# ## Creating a mesh using `gmsh`
+
+# The creation of an unstructured quadrilateral mesh using `gmsh` is driven by a **geometry file**. 
+# There are plenty of possibilities for the user, see the [documentation](https://gmsh.info/doc/texinfo/gmsh.html) and [tutorials](https://gitlab.onelab.info/gmsh/gmsh/tree/master/tutorials).
+
+# To begin, we provide a complete geometry file in this tutorial. After this we give a breakdown
+# of the most important parts required for successful mesh generation that can later be used by the `p4est` library
+# and `Trixi.jl`.
+
+# The associated `NACA6412.geo` file is given below:
+# ```c++
+#  // GMSH geometry script for a NACA 6412 airfoil in a box
+#  // see https://github.com/cfsengineering/GMSH-Airfoil-2D
```

Comment:
This NACA6412 data is for angle of attack 8 right? This should probably be mentioned somewhere

Reply:
Good point, I will double-check that and report it.